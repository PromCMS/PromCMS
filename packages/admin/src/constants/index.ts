export * from './adminLanguages';
export * from './api';
export * from './localization';
export * from './pageUrls';
export * from './misc';
export * from './texts';
