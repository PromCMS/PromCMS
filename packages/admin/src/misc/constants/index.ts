export * from './api'
export * from './localization'
export * from './texts'
export * from './misc'
