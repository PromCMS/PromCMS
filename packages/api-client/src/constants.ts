export const HTTP_STATUS_CODE_FILE_TOO_LARGE = 413;
export const HTTP_STATUS_CODE_UNSUPPORTED_FILE_EXTENSION = 415;
