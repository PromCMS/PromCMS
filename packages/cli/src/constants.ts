import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

export const PACKAGE_ROOT = path.join(__dirname, '..');
export const SCRIPTS_ROOT = path.join(PACKAGE_ROOT, 'scripts');
export const PROJECT_ROOT = path.join(PACKAGE_ROOT, '..', '..');
export const TEMPLATES_ROOT = path.join(PACKAGE_ROOT, 'templates');
