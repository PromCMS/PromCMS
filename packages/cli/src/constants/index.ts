import { join } from 'path';
import { dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));

export const PROJECT_ROOT = join(__dirname, '..', '..', '..');

export const CORE_ROOT = join(PROJECT_ROOT, 'packages', 'core');
export const CLI_ROOT = join(PROJECT_ROOT, 'packages', 'cli');
