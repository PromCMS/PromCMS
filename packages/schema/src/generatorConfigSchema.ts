import { z } from 'zod';
import { databaseConfigSchema } from './databaseConfigSchema.js';
import { projectConfigSchema } from './projectConfigSchema.js';

export const generatorConfigSchema = z.object({
  project: projectConfigSchema,
  database: databaseConfigSchema,
});

export type GeneratorConfig = z.infer<typeof generatorConfigSchema>;
export type GeneratorConfigInput = z.input<typeof generatorConfigSchema>;
