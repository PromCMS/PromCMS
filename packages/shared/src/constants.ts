export const supportedConfigExtensions = [
  'json',
  'js',
  'cjs',
  'mjs',
  'ts',
] as const;
