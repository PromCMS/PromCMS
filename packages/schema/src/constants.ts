import { GeneratorConfigInput } from './generatorConfigSchema.js';
import { FieldPlacements } from './columnType/columnTypeBaseAdminConfigSchema.js';
import { SecurityOptionOptions } from './projectSecurityRoleModelPermissionSchema.js';

export const mockedGeneratorConfig: GeneratorConfigInput = {
  project: {
    name: 'Mocked project',
    url: 'http://localhost',
    security: {
      roles: [
        {
          name: 'Editor',
          modelPermissions: {
            posts: {
              u: SecurityOptionOptions.ALLOW_EVERYTHING,
            },
          },
        },
      ],
    },
  },
  database: {
    connections: [
      {
        name: 'base-connection',
        adapter: 'sqlite',
        dsn: 'sqlite:./database/sq.3',
        user: '',
        password: '',
      },
    ],
    models: [
      {
        tableName: 'posts',
        timestamp: true,
        sorting: true,
        draftable: true,
        preset: 'post',
        title: 'Články',
        admin: {
          icon: 'Archive',
        },
        columns: [
          // Content and title is added as a default
          {
            name: 'description',
            type: 'longText',
            title: 'Popisek',
            admin: {
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
        ],
      },
      {
        tableName: 'pages',
        timestamp: true,
        sorting: true,
        draftable: true,
        admin: { icon: 'Notebook' },
        preset: 'post',
        columns: [
          // Content and title is added as a default
          {
            name: 'showInMenu',
            type: 'boolean',
            title: 'Zobrazit v menu',
            default: false,
            admin: {
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
          {
            name: 'hero_image',
            type: 'file',
            title: 'Úvodní obrázek',
            typeFilter: 'image',
            admin: {
              isHidden: true,
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
          {
            name: 'anotherFile',
            type: 'file',
            title: 'Další soubor',
            admin: {
              isHidden: true,
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
          {
            name: 'someFile',
            type: 'file',
            title: 'Další soubory',
            admin: {
              isHidden: true,
            },
          },
          {
            name: 'excerpt',
            type: 'longText',
            title: 'Krátký popisek',
            localized: true,
            admin: {
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
          {
            name: 'description',
            type: 'longText',
            title: 'Popisek',
            localized: true,
            admin: {
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
        ],
      },
    ],
    singletons: [
      {
        tableName: 'frontpage',

        admin: { icon: 'Archive' },
        preset: 'post',
        columns: [
          {
            name: 'heroImage',
            type: 'file',
            title: 'Úvodní obrázek',
            typeFilter: 'image',
            multiple: true,
            admin: {
              isHidden: true,
              fieldType: 'big-image',
            },
          },
          {
            name: 'image',
            type: 'file',
            title: 'Next image',
            typeFilter: 'image',
            admin: {
              isHidden: true,
              editor: {
                placement: FieldPlacements.ASIDE,
              },
            },
          },
          {
            name: 'openingHours',
            type: 'json',
            title: 'Opening hours test',
            admin: {
              fieldType: 'openingHours',
            },
          },
          {
            name: 'rip',
            type: 'json',
            title: 'Repeater test',
            admin: {
              fieldType: 'repeater',
              editor: {
                placement: FieldPlacements.ASIDE,
              },
              columns: [
                {
                  name: 'name',
                  type: 'string',
                },
                {
                  name: 'value',
                  title: 'value',
                  type: 'string',
                },
                {
                  name: 'unmb',
                  type: 'number',
                },
              ],
            },
          },
          {
            name: 'ripTwo',
            type: 'json',
            title: 'Repeater test',
            admin: {
              fieldType: 'repeater',
              columns: [
                {
                  name: 'name',
                  type: 'string',
                },
              ],
            },
          },
        ],
      },
    ],
  },
};
