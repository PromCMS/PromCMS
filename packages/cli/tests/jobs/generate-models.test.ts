import path from 'path';
import fs from 'fs-extra';
import { describe, beforeEach, it, afterAll, expect } from 'vitest';
import generateModels from '../../src/jobs/generate-models.js';

import { monorepoRoot } from '@prom-cms/shared/dist/internal/constants.js';
import { generatorConfigSchema } from '@prom-cms/schema/dist/generatorConfigSchema.js';
import {
  DatabaseConfig,
  databaseConfigSchema,
} from '@prom-cms/schema/dist/databaseConfigSchema.js';
import { z } from 'zod';
import { MODELS_FOLDER_NAME } from '@constants';

const TEST_FOLDER_PATH = path.join(
  monorepoRoot,
  'node_modules',
  '.prom-cms',
  'test-app'
);
describe('jobs', () => {
  describe('generateModels', () => {
    beforeEach(async () => {
      await fs.remove(TEST_FOLDER_PATH);
      await fs.ensureDir(TEST_FOLDER_PATH);
    });

    afterAll(async () => {
      await fs.remove(TEST_FOLDER_PATH);
    });

    it('should run correctly and generate files', async () => {
      const root = path.join(TEST_FOLDER_PATH, 'modules', 'test');

      await fs.ensureDir(root);

      await generateModels({
        config: {
          project: { name: '', url: '' },
          database: databaseConfigSchema.parse({
            models: {
              first: {
                icon: 'Article',
                admin: { hidden: true },
                columns: { col: { type: 'string', title: 'New' } },
              },
              second: {
                icon: 'Article',
                tableName: 'second_table_name',
                enabled: false,
                columns: { col: { type: 'string', title: 'Newo' } },
              },
            },
          } satisfies z.input<typeof databaseConfigSchema>),
        },
        moduleRoot: root,
      });

      expect(
        await fs.readFile(
          path.join(root, MODELS_FOLDER_NAME, 'First.model.php'),
          { encoding: 'utf8' }
        )
      ).toMatchSnapshot();

      expect(
        await fs.readFile(
          path.join(root, MODELS_FOLDER_NAME, 'Second.model.php'),
          { encoding: 'utf8' }
        )
      ).toMatchSnapshot();
    });
  });
});
