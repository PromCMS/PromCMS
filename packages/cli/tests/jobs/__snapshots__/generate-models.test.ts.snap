// Vitest Snapshot v1

exports[`jobs > generateModels > should run correctly and generate files 1`] = `
"<?php

use PromCMS\\\\Core\\\\Database\\\\Model;
use PromCMS\\\\Core\\\\Database\\\\SingletonModel;
use PromCMS\\\\Core\\\\Database\\\\ModelResult;

class First extends Model
{
  protected string $tableName = 'first';
  protected bool $softDelete = false;

  protected bool $timestamps = true;
  protected bool $translations = true;
  protected static bool $enabled = true;
  protected static array $adminSettings = ['hidden' => true];

  public static array $tableColumns = [
    'col' => [
      'title' => 'New',
      'hide' => false,
      'required' => false,
      'unique' => false,
      'editable' => true,
      'translations' => true,
      'admin' => [
        'isHidden' => false,
        'editor' => ['placement' => 'main'],
        'fieldType' => 'normal',
      ],
      'type' => 'string',
    ],
  ];

  static bool $ignoreSeeding = false;

  static string $modelIcon = 'Article';

  public function getSummary()
  {
    return (object) [
      'isSingleton' => $this instanceof SingletonModel,
      'tableName' => $this->getTableName(),
      'icon' => self::$modelIcon,
      'title' => isset(self::$title) ? self::$title : null,
      'ignoreSeeding' => self::$ignoreSeeding,
      'columns' => static::$tableColumns,
      'hasTimestamps' => $this->hasTimestamps(),
      'hasSoftDelete' => $this->hasSoftDelete(),
      'admin' => self::$adminSettings,
      'enabled' => self::$enabled,
      'ownable' => true,
      'hasOrdering' => false,
      'isDraftable' => false,
      'isSharable' => true,
    ];
  }
}
"
`;

exports[`jobs > generateModels > should run correctly and generate files 2`] = `
"<?php

use PromCMS\\\\Core\\\\Database\\\\Model;
use PromCMS\\\\Core\\\\Database\\\\SingletonModel;
use PromCMS\\\\Core\\\\Database\\\\ModelResult;

class Second extends Model
{
  protected string $tableName = 'second_table_name';
  protected bool $softDelete = false;

  protected bool $timestamps = true;
  protected bool $translations = true;
  protected static bool $enabled = false;
  protected static array $adminSettings = ['hidden' => false];

  public static array $tableColumns = [
    'col' => [
      'title' => 'Newo',
      'hide' => false,
      'required' => false,
      'unique' => false,
      'editable' => true,
      'translations' => true,
      'admin' => [
        'isHidden' => false,
        'editor' => ['placement' => 'main'],
        'fieldType' => 'normal',
      ],
      'type' => 'string',
    ],
  ];

  static bool $ignoreSeeding = false;

  static string $modelIcon = 'Article';

  public function getSummary()
  {
    return (object) [
      'isSingleton' => $this instanceof SingletonModel,
      'tableName' => $this->getTableName(),
      'icon' => self::$modelIcon,
      'title' => isset(self::$title) ? self::$title : null,
      'ignoreSeeding' => self::$ignoreSeeding,
      'columns' => static::$tableColumns,
      'hasTimestamps' => $this->hasTimestamps(),
      'hasSoftDelete' => $this->hasSoftDelete(),
      'admin' => self::$adminSettings,
      'enabled' => self::$enabled,
      'ownable' => true,
      'hasOrdering' => false,
      'isDraftable' => false,
      'isSharable' => true,
    ];
  }
}
"
`;
